src/test/java/com/wojciechkocik/usage/service/CourseUsageServiceTest.java

```diff
@@ -0,0 +1,217 @@
+
+        //Arrange
+        String courseId = fairy.textProducer().randomString(10);
+        String userId = fairy.textProducer().randomString(10);
+        long timeSpent = 5000;
+        ZonedDateTime started = ZonedDateTime.now();
+
+        CourseUsageCreate courseUsageCreate = new CourseUsageCreate();
+        courseUsageCreate.setCourseId(courseId);
+        courseUsageCreate.setUserId(userId);
+        courseUsageCreate.setTimeSpent(timeSpent);
+        courseUsageCreate.setStarted(started);
+
+        CourseUsage courseUsageExpected = new CourseUsage();
+        courseUsageExpected.setTimeSpent(timeSpent);
+        courseUsageExpected.setUserId(userId);
+        courseUsageExpected.setCourseId(courseId);
+        courseUsageExpected.setStarted(started);
+
+        //Act
+        CourseUsage courseUsageActual = courseUsageService.createNewCourseUsage(courseUsageCreate);
+        courseUsageExpected.setId(courseUsageActual.getId());
+
+        //Arrange
+        Assert.assertEquals(courseUsageExpected, courseUsageActual);
+    }
+
+    @Test(expected = ConstraintViolationException.class)
+    public void createNewCourseUsage_whenArgFieldsAreNull_throwsConstraintViolationException() {
+        //Arrange
+        CourseUsageCreate courseUsageCreate = new CourseUsageCreate();
+
+        //Act
+        CourseUsage courseUsageActual = courseUsageService.createNewCourseUsage(courseUsageCreate);
+    }
+
+    @Test
+    public void findDailyUsageForCourse_whenCourseSessionCrossedMidnight_thenRestTimePassedToNextDay() {
+        //Arrange
+        int timeSpentMinutes = 5;
+        String courseId = fairy.textProducer().randomString(10);
+        CourseUsage courseUsageWithTimeCrossedMidnight = new CourseUsage();
+        ZonedDateTime startedOneMinuteBeforeMidnight = ZonedDateTime.parse("2017-03-23T23:59:00.000+01:00[Europe/Warsaw]");
+        String firstSimpleDay = "2017-03-23";
+        String secondSimpleDay = "2017-03-24";
+        courseUsageWithTimeCrossedMidnight.setStarted(startedOneMinuteBeforeMidnight);
+        courseUsageWithTimeCrossedMidnight.setCourseId(courseId);
+        courseUsageWithTimeCrossedMidnight.setUserId(fairy.textProducer().randomString(10));
+        courseUsageWithTimeCrossedMidnight.setTimeSpent(Duration.ofMinutes(timeSpentMinutes).getSeconds()); //one minute in the next day
+
+        courseUsageRepository.save(courseUsageWithTimeCrossedMidnight);
+
+        int responseSizeExpected = 2;
+
+        long firstDayMinutesExpected = 1;
+        long secondDayMinutesExpected = timeSpentMinutes - firstDayMinutesExpected;
+
+        //Act
+        List<DailyUsageResponse> dailyUsagesForCourse = courseUsageService.findDailyUsageForCourse(courseId);
+        int responseSizeActual = dailyUsagesForCourse.size();
+        long firstDayMinutesActual = dailyUsagesForCourse.stream().filter(f -> f.getDateTime().equals(firstSimpleDay))
+                .collect(Collectors.toList()).get(0).getTime();
```

Comment:
If there would be no items on the list, get(0) would throw exception, but test should fail only on at assertions.

You should call findFirst() and validate that the objects exist and then do further assertions.